app/controllers/HomeController.scala

```diff
@@ -34,18 +34,25 @@ object HomeController extends HomeController(ApplicationClient)
 class HomeController(applicationClient: ApplicationClient) extends BaseController(applicationClient) {
   val Withdrawer = "Candidate"
 
+
   val present = CSRSecureAction(ActiveUserRole) { implicit request => implicit cachedData =>
     cachedData.application.map { application =>
+
+
+      def getPhase2Test: Future[Option[Phase2TestGroupWithNames]] = if (application.applicationStatus == ApplicationStatus.PHASE2_TESTS) {
+        applicationClient.getPhase2TestProfile(application.applicationId).map(Some(_))
```

Comment:
if getPhase2TestProfile can return no result then this should be probably reflected in the method signature. So a Future[Option[Something]] rather than Future[Something]